tasks/src/main/scala/dagr/tasks/fgbio/ClipBam.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.tasks.fgbio
+
+import com.fulcrumgenomics.commons.CommonsDef.PathToFasta
+import dagr.tasks.DagrDef.PathToBam
+
+import scala.collection.mutable.ListBuffer
+
+class ClipBam(val input: PathToBam,
+              val output: PathToBam,
+              val ref: PathToFasta,
+              val softClip: Option[Boolean] = None,
+              val autoClipAttributes: Option[Boolean] = None,
+              val readOneFivePrime: Option[Int] = None,
+              val readOneThreePrime: Option[Int] = None,
+              val readTwoFivePrime: Option[Int] = None,
+              val readTwoThreePrime: Option[Int] = None
+             ) extends FgBioTask {
+
+  override protected def addFgBioArgs(buffer: ListBuffer[Any]): Unit = {
+    buffer.append("-i", input)
+    buffer.append("-o", output)
+    buffer.append("-r", ref)
+    softClip.foreach          (s => buffer.append("-s", s))
+    autoClipAttributes.foreach(a => buffer.append("-a", a))
+    readOneFivePrime.foreach  (b => buffer.append("-b", b))
```

Comment:
The `readOneFivePrime` doesn't have a short flag, so would need to be specified with `--read-one-five-prime`.  The same is true for lines 50-52.